### DISCUSSION TITLE
images in docs folder are deleted when rendering to docs

### DISCUSSION THREAD

**Discussion by wiesehahn** [quarto-dev/quarto-cli#5311] [Q&A]
I try to render a predentation/website and want to publish it as Github pages. 

First I had a structure like this, which made problems on githubs side. 
```
├── docs
│   ├── presentation.html
├── resources
│   ├── figures 
├── quarto.yaml
└── presentation.qmd
```

So I tried to put images in the docs folder where the html file is rendered, on githubs side this works, but I have the problem that images in the docs folder are deleted while rendering the html.
```
├── docs
│   ├── presentation.html
│   ├── figures 
├── quarto.yaml
└── presentation.qmd
```

the yaml file looks like 
```
project:
  type: website
  output-dir: docs
```

Is this intended, or is it due to some other settings on my side such as plugins etc.?

**Comment by mcanouil** [comment]
I don't understand the setup, `quarto.yaml` is not a valid file.
You are using "website" project but for a presentation as it appears.

From what I understood, you did not specify `resources` to tell Quarto there are some "external" resources, see <https://quarto.org/docs/websites/website-tools.html#site-resources>.

**Reply by wiesehahn** [reply]
I am not sure how the "website" project sneaked in there. And I am still confused what goes in the .yaml and what goes in the header of .qmd. 

So with the goal to render a reveal presentation and publish it as github pages of that project, how would a good setup look?

My current attempt looks as following:
presentation.qmd should render to docs/presentation.html and figures should be read from docs/figures

```
├── docs
│   ├── presentation.html
│   ├── figures 
├── quarto.yml
└── presentation.qmd
```

with .yaml
```
project:
  output-dir: docs
```

This basically works, figures are not deleted anymore, but libs are created in presentation_files/libs while presentation.html expects them at docs/presentation_files/libs

**Reply by mcanouil** [reply]
Without an actual example, it's really hard to understand or even try to fiddle with something out of nothing.
Could you share something such as a repository where you already tried something?

**Reply by wiesehahn** [reply]
sure, https://github.com/wiesehahn/lidar_im_forst_vortrag is the repo. 
On the first run libs were created under libs are created in presentation_files/libs, rerunning seems to keep them under docs. 

It would be nice to know if the structure I chose is the way to go here or if I messed it up in some way. Thanks

**Reply by mcanouil** [reply]
`output-dir` is really a Quarto project option and is really working as expected in books/websites projects.
If you only want to render/deploy your presentation in GitHub Pages, the cleanest/simplest way, is to render/publish your presentation to the `gh-branch`, see <https://quarto.org/docs/publishing/github-pages.html>.
If you do not want to automate, you can simply run. `quarto publish gh-pages` locally each time you want to update the deployed HTML file.
